File: backend/scripts/check_db.js
import Database from 'better-sqlite3';
import path from 'path';

const dbPath = path.join(process.cwd(), 'data', 'files.db');
const db = new Database(dbPath);

console.log('--- Files Table ---');
const files = db.prepare('SELECT id, name, master_key_encrypted FROM files LIMIT 10').all();
files.forEach(f => {
    console.log(`ID: ${f.id}, Name: ${f.name}, MasterKey: ${f.master_key_encrypted?.substring(0, 10)}... (Length: ${f.master_key_encrypted?.length})`);
});

console.log('--- Chunks Table ---');
const chunks = db.prepare('SELECT id, file_id, chunk_index, iv, auth_tag FROM chunks LIMIT 10').all();
chunks.forEach(c => {
    console.log(`ID: ${c.id}, FileID: ${c.file_id}, IV: ${c.iv?.substring(0, 10)} (Length: ${c.iv?.length}), Tag: ${c.auth_tag?.substring(0, 10)} (Length: ${c.auth_tag?.length})`);
});

File: backend/scripts/check_db.py
import sqlite3
import os

db_path = 'data/files.db'
if not os.path.exists(db_path):
    print(f"Error: {db_path} not found")
    exit(1)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

try:
    print("--- Files Table Sample (id, name, created_at, key_len) ---")
    cursor.execute('SELECT id, name, created_at, length(master_key_encrypted) FROM files ORDER BY created_at DESC LIMIT 20')
    for row in cursor.fetchall():
        print(row)

    print("\n--- Chunks Table Sample (file_id, iv_len, tag_len) ---")
    cursor.execute('SELECT file_id, length(iv), length(auth_tag) FROM chunks LIMIT 20')
    for row in cursor.fetchall():
        print(row)
except Exception as e:
    print(f"SQL Error: {e}")

conn.close()
